framework/include/utils/MooseTypes.h

```diff
@@ -185,6 +185,12 @@ enum LineSearchType
 #endif
 };
 
+enum RegistrationType
+{
+  APPLICATION,
+  OBJECTS
+};
+
```

Comment:
This isn't used in the current PR, but I think perhaps we should think forward and possibly add it to the extern "C" function for more flexibility since this API change will need to be applied to _all_ applications. I'll explain more in the AnimalApp.C file...